extensions/wikia/CoppaTool/RenameIPLogFormatter.class.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+/**
+ * @author: Władysław Bodzek
+ *
+ * A helper class for the User rename tool
+ * @copyright (C) 2010, Wikia Inc.
+ * @licence GNU General Public Licence 3.0 or later
+ */
+class RenameIPLogFormatter {
+	static public function getCommunityUser( $name, $noRedirect = false ) {
+		if ( is_int( $name ) )
+			$name = User::whoIs( $name );
+		$title = GlobalTitle::newFromText( $name, NS_USER, COMMUNITY_CENTRAL_CITY_ID );
+		return Xml::element( 'a', [ 'href' => $title->getFullURL(
+			$noRedirect ? 'redirect=no' : ''
+		) ], $name, false );
+	}
+
+	static protected function getCommunityTask( $taskId ) {
+		$title = GlobalTitle::newFromText( 'Tasks/log', NS_SPECIAL, COMMUNITY_CENTRAL_CITY_ID );
+		return Xml::element(
+			'a',
+			[
+				'href' => $title->getFullURL( [ 'id' => $taskId ] ),
+			],
+			"#{$taskId}",
+			false
+		);
+	}
+
+	static public function getCityLink( $cityId ) {
+		global $wgCityId, $wgSitename;
+		$domains = WikiFactory::getDomains( $cityId );
+		if ( $wgCityId == $cityId ) {
+			// Hack based on the fact we should only ask for current wiki's sitename
+			$text = $wgSitename;
+		} else {
+			// The fallback to return anything
+			$text = "[" . WikiFactory::IDtoDB( $cityId ) . ":{$cityId}]";
+		}
+		if ( !empty( $domains ) ) {
+			$text = Xml::tags( 'a', [ "href" => "http://" . $domains[0] ], $text );
+		}
+		return $text;
+	}
+
+	static public function start( $requestor, $oldIPAddress, $newIPAddress, $reason, $tasks = [ ] ) {
+		foreach ( $tasks as $k => $v ) {
+			$tasks[$k] = self::getCommunityTask( $v );
+		}
+
+		$text = wfMessage( 'coppatool-info-started',
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$text". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

[project-id]: wikia:app